mmpose/datasets/datasets/hand/interhand3d_dataset.py

```diff
@@ -0,0 +1,534 @@
+                    pred_left_root_img[:, 2] += item['abs_depth'][0] + pred[
+                        'rel_root_depth']
+                    pred_left_root_cam = self._pixel2cam(
+                        pred_left_root_img, item['focal'], item['princpt'])
+
+                    pred_right_root_img = np.array(
+                        pred['keypoints'][20], dtype=np.float32)[None, :]
+                    pred_right_root_img[:, 2] += item['abs_depth'][0]
+                    pred_right_root_cam = self._pixel2cam(
+                        pred_right_root_img, item['focal'], item['princpt'])
+
+                    preds_rel_root.append(pred_left_root_cam -
+                                          pred_right_root_cam)
+                    gts_rel_root.append(
+                        [item['joints_cam'][41] - item['joints_cam'][20]])
+                else:
+                    rel_root_masks.append(False)
+                    preds_rel_root.append([[0., 0., 0.]])
+                    gts_rel_root.append([[0., 0., 0.]])
+
+            if 'MPJPE' in metrics:
+                pred_joint_coord_img = np.array(
+                    pred['keypoints'], dtype=np.float32)
+                gt_joint_coord_cam = item['joints_cam'].copy()
+
+                pred_joint_coord_img[:21, 2] += item['abs_depth'][0]
+                pred_joint_coord_img[21:, 2] += item['abs_depth'][1]
+                pred_joint_coord_cam = self._pixel2cam(pred_joint_coord_img,
+                                                       item['focal'],
+                                                       item['princpt'])
+
+                pred_joint_coord_cam[:21] -= pred_joint_coord_cam[20]
+                pred_joint_coord_cam[21:] -= pred_joint_coord_cam[41]
+                gt_joint_coord_cam[:21] -= gt_joint_coord_cam[20]
+                gt_joint_coord_cam[21:] -= gt_joint_coord_cam[41]
+
+                preds_joint_coord_cam.append(pred_joint_coord_cam)
+                gts_joint_coord_cam.append(gt_joint_coord_cam)
+
+                mask = (np.array(item['joints_3d_visible'])[:, 0]) > 0
+
+                if item['hand_type'].all():
+                    single_masks.append(
+                        np.zeros(self.ann_info['num_joints'], dtype=bool))
+                    interacting_masks.append(mask)
+                    all_masks.append(mask)
+                else:
+                    single_masks.append(mask)
+                    interacting_masks.append(
+                        np.zeros(self.ann_info['num_joints'], dtype=bool))
+                    all_masks.append(mask)
+
+            if 'Handedness_acc' in metrics:
+                pred_hand_type = np.array(pred['hand_type'], dtype=int)
+                preds_hand_type.append(pred_hand_type)
+                gts_hand_type.append(item['hand_type'])
+                hand_type_masks.append(item['hand_type_valid'] > 0)
+
+        gts_rel_root = np.array(gts_rel_root)
```

Comment:
Make the datatype explict

Reply:
E.g., there are some variables with `mask` in their names. Then one may wonder their types: bool? int? float32?

Reply:
The type can be changed if the append operation is not careful. e.g.

![image](https://user-images.githubusercontent.com/9464825/108513235-42539b80-72fd-11eb-9de0-995ce3db564a.png)